src/radical/pilot/task.py

```diff
@@ -461,6 +472,29 @@ def metadata(self):
         return self._descr.metadata
 
 
+    # --------------------------------------------------------------------------
+    #
+    @property
+    def info(self):
+        """The metadata field of the task's description."""
+        return self._info
+
+    def _set_info(self, info):
+        self._info = info or ''
+        self._info_evt.set()
+
+    def wait_info(self, timeout=None):
+        self._info_evt.wait(timeout=timeout)
+
+        import pprint
+        pprint.pprint(self._info)
+
+        if self._info['0'] is None:
```

Comment:
and `self._info` might contain `''` or `str` or `dict`

Reply:
You know, I probably should change the `service_info` back to a single string.  I don't think the dict approach provides value, let's see if a use case comes up which actually needs that...